### PR TITLE
FPGA: Reduce test size from 64 to 2 for memory_attributes simulator target

### PR DESCRIPTION
**Pull request by intel-liudean**
# Existing Sample Changes
## Description
Currently this test takes unreasonably long (hours) for Windows simulator target. Reducing the test size helps to reduce the time taken to under 10 minutes.

## Type of change
Reduce test size

## How Has This Been Tested?
Reducing the test size should be a safe change.
I have also test the simulator target on both Windows and Linux for my change.

### CODE REVIEW

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@ using namespace sycl;
 constexpr size_t kRows = 8;
 constexpr size_t kVec = 4;
 constexpr size_t kMaxVal = 512;
-constexpr size_t kNumTests = 64;
+constexpr size_t kNumTests = 2;
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp:17]: Should this restriction only be applied to the simulator flow?

Comment (intel-liudean) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp:17]: This is a great idea! I was not aware that this is possible.
I wonder do you know how to do this in the code?

Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp:17]:
```suggestion
#if defined (FPGA_SIMULATOR)
constexpr size_t kNumTests = 2;
#else
constexpr size_t kNumTests = 64;
#endif
```

Comment (intel-liudean) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp:17]: Thank you Yohann, let me commit this and do the testing again locally

Comment (intel-liudean) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/memory_attributes/src/memory_attributes.cpp:18]: Testing looks good

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**